### PR TITLE
fix: log api errors with status and message

### PR DESCRIPTION
**Pull request by valmack**
### Motivation

Status code in the logs were always 200 when the actual response sent back a 5xx.

### Changes

- Send response first then produce the http log so that `res.statusCode` and `res.statusMessage` are accurate
- Also only log usage stats when healthcheck fails to reduce noise of healthcheck logs

### CODE REVIEW

File: packages/cli/src/ceramic-daemon.ts [hunk 1]
```diff
@@ -212,12 +215,12 @@ class CeramicDaemon {
     const stats = `maxHealthyCpu=${this.maxHealthyCpu} cpuUsage=${cpuUsage} freeCpu=${freeCpu} maxHealthyMemory=${this.maxHealthyMemory} memoryUsage=${memUsage} freeMemory=${freeMemory}`
 
     if (cpuUsage > this.maxHealthyCpu || memUsage > this.maxHealthyMemory) {
-      this.logger.error('Ceramic failed a healthcheck')
-      res.status(503).send('Insufficient resources')
+      this.logger.debug(stats)
```
Comment (stbrody) [inline packages/cli/src/ceramic-daemon.ts:218]: hmm... it still feels useful to log an error if we actually fail a healthcheck I think?  I think this should probably still be logger.error, with a message that we failed a healthcheck, *and* with the stats included.

File: packages/cli/src/ceramic-daemon.ts [hunk 2]
```diff
@@ -95,12 +95,15 @@ class CeramicDaemon {
...
+    // next is required in function signature
     app.use((err: Error, req: Request, res: Response, next: NextFunction): void => {
-      const requestStart = Date.now()
-      const httpLog = this._buildHttpLog(requestStart, req, res)
-      const logString = JSON.stringify(httpLog)
-      this.logger.error(logString)
-      next(err)
+      this.logger.error(err)
+      if (res.statusCode < 300) { // 2xx indicates error has not yet been handled
+        res.status(500)
```
Comment (stbrody) [inline packages/cli/src/ceramic-daemon.ts:102]: I guess I don't really understand enough about how HTTP status codes are used.  Why wouldn't we just use `res.statusCode` here?  Aren't we losing information by not using the statusCode and just always using 500?

**Review by stbrody — COMMENTED** [review]
I think this review may be too far out of area of expertise for me to give meaningful input.  I left a comment about a log message, but I think I'm going to defer to Joel to review the changes to the actual HTTP handling

File: packages/cli/src/ceramic-daemon.ts (hunk 2 above)
Comment (oed) [reply packages/cli/src/ceramic-daemon.ts:102]: I assume this handler will only get called if there is an error elsewhere. Then if the status code has already been set to 200 we should change it to 500 in order to reflect that there was an error.

**Review by oed — COMMENTED** [review]
This seems good to me if my assumption below is correct :)

File: packages/cli/src/ceramic-daemon.ts (hunk 2 above)
Comment (valmack) [reply packages/cli/src/ceramic-daemon.ts:102]: Yes exactly @oed. Also @stbrody this is where the "Ceramic failed a healthcheck" will get logged

File: packages/cli/src/ceramic-daemon.ts (hunk 1 above)
Comment (valmack) [reply packages/cli/src/ceramic-daemon.ts:218]: Other comment addresses this. so yes the message will get logged as an error and the stats will get logged as debug.

**Pull request merged**